### DISCUSSION TITLE
Webhook deployments from git commits should respect [skip ci]

### DISCUSSION THREAD

**Discussion by sisou** [coollabsio/coolify#4358] [Improvement Requests]
### Error Message and Logs

For CIs it has been established as a standard to be able to put `[skip ci]` somewhere in a commit message to tell the CI to _not_ run with this commit.

I noticed earlier that deployments by webhook for, in my case, Github pushes still trigger deployments in Coolify.

I think they should be skipped when `[skip ci]` is found in the commit message.

It's also ok to introduce another keyword.

### Steps to Reproduce

1. Push a commit to a auto-deployed repo with `[skip ci]` in the commit message
2. Observe that any configured Github Actions do not run for this commit
3. Observe that Coolify still deploys the commit

### Coolify Version

v4.0.0-beta.370

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

Ubuntu 22.04.5 LTS

**Comment by luco** [comment]
Not resolved at [v4.0.0-beta.416](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.416) as well.

**Comment by whitespacecode** [comment]
Yes please!

**Comment by narduin** [comment]
This would be really great, yes.

**Comment by sovetski** [comment]
good idea !

**Comment by light-merlin-dark** [comment]
First, thank you so much for Coolify and all the brilliance that goes into this platform.  I love it and it's a real blessing.  Second, yes this would be a wonderful feature.

**Comment by stevenirby** [comment]
This would be really great to have! But I would recommend doing this how Netlify does it: https://docs.netlify.com/deploy/manage-deploys/manage-deploys-overview/

`[skip ci]` to skip the github ci or wherever your code is hosted.
`[skip coolify]` to skip coolify building the code

**Comment by ton77v** [comment]
+1 it's not nice to have a bunch of deployments auto-triggered by e.g. changelog updates and other automated stuff

**Reply by jpmadrigal07** [reply]
Yeah, really annoying

**Comment by ffflorian** [comment]
Seems like this was done in https://github.com/coollabsio/coolify/pull/9861 🎉

**Comment by ton77v** [comment]
Heads-up for anyone still hitting this: the per-commit case for GitHub **PR preview** deploys survived #9861. `ProcessGithubPullRequestWebhook` passes only the PR *title* into `shouldSkipDeployAny`, and `pull_request` / `synchronize` payloads carry no commit messages — so a `[skip ci]` **commit** still triggers a preview (only the PR title is honored). Filed a focused follow-up: #10870.
